cassandra-four-zero-bridge/src/main/java/org/apache/cassandra/bridge/SSTableWriterImplementation.java

```diff
@@ -69,94 +56,56 @@ public SSTableWriterImplementation(String inDirectory,
-        this.sstableWatcher = new SSTableWatcher(sstableWatcherDelaySeconds);
     }
 
-    private class SSTableWatcher implements Closeable
+    private static IPartitioner determineSupportedPartitioner(String partitioner)
     {
-        // The TOC component is the last one flushed when finishing a SSTable.
-        // Therefore, it monitors the creation of the TOC component to determine the creation of SSTable
-        private static final String TOC_COMPONENT_SUFFIX = "-TOC.txt";
-        private static final String GLOB_PATTERN_FOR_TOC = "*" + TOC_COMPONENT_SUFFIX;
-
-        private final ScheduledExecutorService sstableWatcherScheduler;
-        private final Set<SSTableDescriptor> knownSSTables;
-
-        SSTableWatcher(long delaySeconds)
-        {
-            ThreadFactory tf = ThreadUtil.threadFactory("SSTableWatcher-" + outputDir.getFileName().toString());
-            this.sstableWatcherScheduler = Executors.newSingleThreadScheduledExecutor(tf);
-            this.knownSSTables = new HashSet<>();
-            sstableWatcherScheduler.scheduleWithFixedDelay(this::listSSTables, delaySeconds, delaySeconds, TimeUnit.SECONDS);
-        }
+        return partitioner.toLowerCase().contains("random")
+               ? new RandomPartitioner()
+               : new Murmur3Partitioner();
+    }
 
-        private void listSSTables()
-        {
-            try (DirectoryStream<Path> stream = Files.newDirectoryStream(outputDir, GLOB_PATTERN_FOR_TOC))
-            {
-                HashSet<SSTableDescriptor> newlyProducedSSTables = new HashSet<>();
-                stream.forEach(path -> {
-                    String baseFilename = path.getFileName().toString().replace(TOC_COMPONENT_SUFFIX, "");
-                    SSTableDescriptor sstable = new SSTableDescriptor(baseFilename);
-                    if (!knownSSTables.contains(sstable))
-                    {
-                        newlyProducedSSTables.add(sstable);
-                    }
-                });
-
-                if (!newlyProducedSSTables.isEmpty())
-                {
-                    knownSSTables.addAll(newlyProducedSSTables);
-                    producedSSTablesListener.accept(newlyProducedSSTables);
-                }
-            }
-            catch (IOException e)
-            {
-                LOGGER.warn("Fails to list SSTables", e);
-            }
-        }
+    private void onSSTablesProduced(Collection<SSTableReader> sstables)
+    {
+        Objects.requireNonNull(producedSSTablesListener, "producedSSTablesListener is not set");
+        Set<SSTableDescriptor> sstableDescriptors = sstables
+                                                    .stream()
+                                                    .map(sstable -> {
+                                                        String baseFilename = baseFilename(sstable.descriptor);
+                                                        // TODO: for now, the sstableReader is closed immediately,
```

Comment:
should we do this as part of this PR? I'd argue that this is desirable. Any reason to not do it here?

Reply:
I left the todo because the change would be more invasive and I feel they are unrelated. The step does not simply re-read the data back, but also calculates digests, etc. Some refactoring will be required.

Reply:
maybe worth filing a jira to keep better track of this change?